### PR TITLE
dir_scan gatherer docs

### PR DESCRIPTION
**Pull request by CDimonaco**
As title.

### CODE REVIEW

File: guides/gatherers.md [hunk 1]
```diff
@@ -242,6 +243,43 @@ Example output (in Rhai):
 #{ nodeid: 2, ring0_addr: "10.80.1.12" };
 ```
 
+### dir_scan
+
+**Argument required**: Yes, a glob pattern for directory/files scanning
```
Comment (arbulu89) [inline guides/gatherers.md:248]: I think this line should be simply `yes` (small case).

File: guides/gatherers.md [hunk 2]
```diff
@@ -242,6 +243,43 @@ Example output (in Rhai):
...
+
+Example argument:
```
Comment (arbulu89) [inline guides/gatherers.md:250]: Let's put `Example arguments` after the main description (`This gatherer allows...`).

Something like:
```
...information associated to each file.

The provided argument is a glob pattern for directory/files scanning.

Example arguments:
- `/usr/sap/[A-Z][A-Z0-9][A-Z0-9]/ERS[0-9][0-9]`
- `/etc/polkit-1/rules.d/[0-9][0-9]-SAP[A-Z][A-Z0-9][A-Z0-9]-[0-9][0-9].rules`
```

File: guides/gatherers.md [hunk 3]
```diff
@@ -242,6 +243,43 @@ Example output (in Rhai):
...
+
+- `/usr/sap/[A-Z][A-Z0-9][A-Z0-9]/ERS[0-9][0-9]`
+- `etc/polkit-1/rules.d/[0-9][0-9]-SAP[A-Z][A-Z0-9][A-Z0-9]-[0-9][0-9].rules`
```
Comment (arbulu89) [inline guides/gatherers.md:253]: `/` missing at the beginning of path

File: guides/gatherers.md [hunk 4]
```diff
@@ -242,6 +243,43 @@ Example output (in Rhai):
...
+    gatherer: dir_scan
+    argument: "/usr/sap/[A-Z][A-Z0-9][A-Z0-9]/ERS[0-9][0-9]"
+```
+
+Example output (in Rhai):
+
+```ts
+  [
+    #{
+      "name": "/etc/sap/X/Y",
```
Comment (arbulu89) [inline guides/gatherers.md:272]: Let's put something more real based on the previous example specification like:
```
"name": "/etc/sap/PRD/ERS01",
...
"name": "/etc/sap/QAS/ERS02",
```

File: guides/gatherers.md [hunk 5]
```diff
@@ -242,6 +243,43 @@ Example output (in Rhai):
...
+      "owner": "trento",
+      "group": "trento"
+    },
+     #{
+      "name": "/etc/sap/X/Z",
+      "owner": "trento",
+      "group": "trento"
+    },
+  ]
+```
```
Comment (arbulu89) [inline guides/gatherers.md:282]: Let's add a new line here

**Review by arbulu89 — CHANGES_REQUESTED** [review]
The content is good, but I think some changes in order of things is needed

File: guides/gatherers.md [hunk 6]
```diff
@@ -23,6 +23,7 @@ Here's a collection of built-in gatherers, with information about how to use the
 | [`cibadmin`](#cibadmin)                 | [trento-project/agent/../gatherers/cibadmin.go](https://github.com/trento-project/agent/blob/main/internal/factsengine/gatherers/cibadmin.go)               |
 | [`corosync.conf`](#corosyncconf)        | [trento-project/agent/../gatherers/corosyncconf.go](https://github.com/trento-project/agent/blob/main/internal/factsengine/gatherers/corosyncconf.go)       |
 | [`corosync-cmapctl`](#corosync-cmapctl) | [trento-project/agent/../gatherers/corosynccmapctl.go](https://github.com/trento-project/agent/blob/main/internal/factsengine/gatherers/corosynccmapctl.go) |
+| [`dir_scan`](#dir_scan)                       | [trento-project/agent/../gatherers/dir_scan.go](https://github.com/trento-project/agent/blob/main/internal/factsengine/gatherers/dir_scan.go)
```
Comment (arbulu89) [inline guides/gatherers.md:26]: Can we align the `|` ?
The `|` at the end is missing as well

File: guides/gatherers.md [hunk 7]
```diff
@@ -242,6 +243,44 @@ Example output (in Rhai):
 #{ nodeid: 2, ring0_addr: "10.80.1.12" };
 ```
 
+### dir_scan
+
+**Argument required**: Yes
+
+This gatherer allows to scan directories due to a glob pattern provided as argument.
```
Comment (arbulu89) [inline guides/gatherers.md:250]: I would change `... directories due to a` by `directories with a glob pattern`

File: guides/gatherers.md [hunk 8]
```diff
@@ -242,6 +243,44 @@ Example output (in Rhai):
...
+- `/usr/sap/[A-Z][A-Z0-9][A-Z0-9]/ERS[0-9][0-9]`
+- `/etc/polkit-1/rules.d/[0-9][0-9]-SAP[A-Z][A-Z0-9][A-Z0-9]-[0-9][0-9].rules`
+
+Example specification:
+
+```yaml
+facts:
+  - name: dir_scan
+    gatherer: dir_scan
+    argument: "/usr/sap/[A-Z][A-Z0-9][A-Z0-9]/ERS[0-9][0-9]"
```
Comment (arbulu89) [inline guides/gatherers.md:264]: Should we comment that quotes are mandatory? (if they are)

**Review by arbulu89 — APPROVED** [review]
Thank you!
Left some last leftover comments

File: guides/gatherers.md (hunk 6 above)
Comment (CDimonaco) [reply guides/gatherers.md]: done

File: guides/gatherers.md (hunk 7 above)
Comment (CDimonaco) [reply guides/gatherers.md:250]: done!

File: guides/gatherers.md [hunk 9]
```diff
@@ -242,6 +243,44 @@ Example output (in Rhai):
...
+```
+
+Example output (in Rhai):
+
+```ts
+  [
+    #{
+      "name": "/etc/sap/PRD/ERS01",
```
Comment (arbulu89) [inline guides/gatherers.md:272]: This should be `/usr/sap` not` /etc/sap`

**Pull request merged**